### ISSUE TITLE
Topological map visualization in Tensorboard

### ISSUE THREAD

**Issue by alex-petrenko** [alex-petrenko/landmark-exploration#2]

**Comment by pelillian** [comment]
cbd11b9023908ce4c2ad1fec27517c22079b0a98

**Comment by alex-petrenko** [comment]
Will it write to the same summary file as the rest of the algorithms?
I think you're creating a new summary by doing this.

**Comment by pelillian** [comment]
OK. We could just use TensorboardX for everything, or I could use the regular file writer. I'd be willing to refactor everything to use tensorboardX as it may be more powerful

**Comment by alex-petrenko** [comment]
Whichever is faster.
If you feel strong, then you can go ahead and do it, but keep in mind that you will need to do it in at least 3 algorithms that we have in the repo, and not break anything (it's ICM, PPO, TMAX). And it should be as convenient to use with Tensorflow as regular tensorboard.

Honestly, I don't think we need tensorboardX just for that

**Comment by pelillian** [comment]
OK, I'll see if I can use regular tensorboard then

**Comment by alex-petrenko** [comment]
The current implementation is not correct.
First of all, you're passing `graph` variable directly to the visualization function which causes it to crash. I think you meant `graph.nx_graph`.
Second, which is more important, you're creating a new summary writer for _each_ training step, which in turn creates a new events file every time.
After only a few minutes of training I have like 50 event files in my tensorboard, which completely messes up the summaries and prevents all the other summaries from being displayed (I mean, rewards, etc.)
I don't know, maybe it's ok to create one more writer, but definitely not a new one every step.

Bottom line: please don't commit code unless you tested it. To test this function you should have verified that all the project tests pass (which involve short training session on TMAX) and ideally also run the training script at least once.

**Comment by alex-petrenko** [comment]
I think we can close it now. Works fine for me
